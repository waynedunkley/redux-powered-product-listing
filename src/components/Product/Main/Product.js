import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  ProductImage,
  ProductTitle,
  ProductDescription,
  ProductPrice,
} from "components"
import divStyles from "./Product.style"

const StyledDiv = styled.div`
  ${divStyles};
`

const propTypes = {
  product: PropTypes.object.isRequired,
}

const Product = ({ product }) => (
  <StyledDiv>
    <ProductImage url={product.image.url} alt={product.image.alt} />
    <ProductTitle text={product.title} />
    <ProductDescription text={product.description} />
    <ProductPrice price={product.price} />
  </StyledDiv>
)

Product.propTypes = propTypes

export default Product
