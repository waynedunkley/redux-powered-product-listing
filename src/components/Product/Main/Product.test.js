import React from "react"
import { shallow } from "enzyme"
import Product from "./Product"

it("Product matches snapshot", () => {
  const data = {
    title: "Product title",
    description: "Product description",
    price: "$99.99",
    image: {
      url: "path/to/img",
      alt: "alt text",
    },
  }
  const wrapper = shallow(<Product product={data} />)
  expect(wrapper).toMatchSnapshot()
})
