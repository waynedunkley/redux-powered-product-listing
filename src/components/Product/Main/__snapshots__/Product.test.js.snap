// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Product
    product={
        Object {
            "description": "Product description",
            "image": Object {
              "alt": "alt text",
              "url": "path/to/img",
            },
            "price": "$99.99",
            "title": "Product title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ProductImage
          alt="alt text"
          url="path/to/img"
/>,
        <styled.div>
          <ProductTitle
                    text="Product title"
          />
          <ProductDescription
                    text="Product description"
          />
          <ProductPrice
                    price="$99.99"
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "alt text",
          "url": "path/to/img",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ProductTitle
              text="Product title"
/>,
            <ProductDescription
              text="Product description"
/>,
            <ProductPrice
              price="$99.99"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "text": "Product title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "text": "Product description",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "price": "$99.99",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ProductImage
            alt="alt text"
            url="path/to/img"
/>,
          <styled.div>
            <ProductTitle
                        text="Product title"
            />
            <ProductDescription
                        text="Product description"
            />
            <ProductPrice
                        price="$99.99"
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "alt text",
            "url": "path/to/img",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ProductTitle
                text="Product title"
/>,
              <ProductDescription
                text="Product description"
/>,
              <ProductPrice
                price="$99.99"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "Product title",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "Product description",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "price": "$99.99",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
